frontend/packages/console-plugin-sdk/README.md

```diff
@@ -0,0 +1,234 @@
+
+``` json
+// File: package.json
+
+{
+  "name": "@console/demo-plugin",
+  "version": "0.0.0-fixed",
+  "description": "Demo plugin for Console web application",
+  "private": true,
+  "dependencies": {
+    "@console/plugin-sdk": "0.0.0-fixed",
+    "@console/shared": "0.0.0-fixed"
+  },
+  "devDependencies": {
+    "@console/internal-integration-tests": "0.0.0-fixed"
+  },
+  "consolePlugin": {
+    "entry": "src/plugin.tsx",
+    "integrationTests": {
+      "demo": [
+        "@console/internal-integration-tests/tests/base.scenario.ts",
+        "integration-tests/tests/demo.scenario.ts"
+      ]
+    }
+  }
+}
+```
+
+## Plugin Entry Module (`.../src/plugin.tsx`)
+
+The Plugin Entry Module is defined in `package.json` in the `consolePlugin.entry` variable,
+relative to the plugin root directory.
+
+``` json
+// File: package.json
+
+{
+  ...
+  "consolePlugin": {
+    "entry": "src/plugin.tsx",
+    ...
+  }
+}
+```
+
+The Plugin Entry Module will export an array of plugin objects of type `Plugin<ConsumedExtensions>`. Each `Plugin` object have a type
+declaring the plugin capabilities and options (for example `Page/Route`, `Dashboards/Card` types).
+
+``` es6
+// File: src/plugin.tsx
+
+const plugin: Plugin<ConsumedExtensions> = [
+  ...
+  {
+    type: 'Dashboards/Tab',
+    properties: {
+      id: 'persistent-storage',
+      title: 'Persistent Storage',
+    },
+  },
+  ...
+];
+
+export default plugin;
+```
+
+See the [demo plugin](frontend/packages/console-demo-plugin) for available Plugin extension points.
+
+For better type checking and code completion, use a type parameter that represents the union of all the extension types consumed by the plugin.
+
+``` es6
+// don't do this
+const plugin: Plugin<any> = [ /* stuff */ ];
+// do this
+const plugin: Plugin<FooExtension | BarExtension> = [ /* stuff */ ];
+// or better yet, this
+type ConsumedExtensions = FooExtension | BarExtension;
+const plugin: Plugin<ConsumedExtensions> = [ /* stuff */ ];
+```
+
+## Monorepo Architecture
+
+Console plugins are 1st class citizens of Console monorepo and therefore reuse the same testing (Jest, Protractor, etc.) infrastructure and conventions.
+See Tests and OWNERS sections below for details.
+
+## Unit Tests
+
+The jest framework is used for unit tests. It will automatically run any file located in a directory named `.../__tests__` or whose name ends with a `.spec.(ts|tsx|js|jsx)` suffix.
+
+``` bash
+  .
+  ├── my-plugin
+  |   ├── src
+  |   |   ├── __tests__
+  |   |   |   ├── some-test.spec.ts
+  |   |   |   └── ...
+  |   |   └── ...
+  |   └── ...
+  └── ...
+```
+
+## Integration Tests
+
+Integration tests use [protractor](http://www.protractortest.org) and must be defined in the `consolePlugin` map, in the plugins `package.json`.
+Test path is relative the plugin root directory. It is possible to include common tests defined in `@console/internal-integration-tests`,
```

Comment:
"Test path is relative"

"The test path is relative"

Also, you can remove the comma after `@console/internal-integration-tests`.